src/app/qgsmaptooladdring.cpp

```diff
@@ -29,6 +29,8 @@ QgsMapToolAddRing::QgsMapToolAddRing( QgsMapCanvas* canvas )
     : QgsMapToolCapture( canvas, QgisApp::instance()->cadDockWidget(), QgsMapToolCapture::CapturePolygon )
 {
   mToolName = tr( "Add ring" );
+  connect( QgisApp::instance(), SIGNAL( newProject() ), this, SLOT( stopCapturing() ) );
+  connect( QgisApp::instance(), SIGNAL( projectRead() ), this, SLOT( stopCapturing() ) );
```

Comment:
@m-kuhn wrote:
> What's your opinion about moving this to the `QgsMapToolCapture` constructor?

@SebDieBln commented on 16 Dec 2015 
> Since `QgisApp` is not reachable from the base class (it's in GUI), I added the handlers to the individual tools (they are in APP).

Is there any workaround for this problem?